### PR TITLE
Update types

### PR DESCRIPTION
**Pull request by GrayStrider**
Work on #37 
typecheck & tests ☑️

### CODE REVIEW

**Comment by GrayStrider** [comment]
I'll review this PR to have as little conflicts with upstream as possible

File: app/containers/App/types.d.ts [hunk 1]
```diff
@@ -1,18 +1,19 @@
...
   readonly loading: boolean;
-  readonly error: object | boolean;
+  readonly error?: object | boolean;
   readonly currentUser: string;
   readonly userData: UserData;
 }
 
 interface UserData {
-  readonly repositories: object[] | boolean; // too many fields. Won't declare them all
+  readonly repos?: Repo[]; // too many fields. Won't declare them all
```
Comment (Can-Sahin) [inline app/containers/App/types.d.ts]: regarding to #48 I would avoid renaming things. Adding types etc is really nice but I wanna stick to original code as much as possible with namings and implementation.

**Comment by Can-Sahin** [comment]
Apart from my comment on renaming it looks nice. Didn't go through it detailly though. Can you let me know when you are complete (considering the things we discuss at #48) and I can merge. Could you check again for minimal non-ts changes. Thanks alot

**Comment by GrayStrider** [comment]
Yeah, sorry for the delay, I'm still working on it! I'll remove the renaming for one, yes. @Can-Sahin 
I'll let you know when it's ok to merge.

File: app/components/List/index.tsx [hunk 2]
```diff
@@ -2,10 +2,11 @@ import * as React from 'react';
 
 import Ul from './Ul';
 import Wrapper from './Wrapper';
+import { Repo } from '../../containers/RepoListItem/types';
 
 interface Props {
-  component: any;
-  items?: any[];
+  component: React.ComponentType<any>;
```
Comment (GrayStrider) [inline app/components/List/index.tsx:8]: might need some work, but better than any

File: app/components/ReposList/tests/index.test.tsx [hunk 3]
```diff
@@ -31,7 +32,7 @@ describe('<ReposList />', () => {
         error: false,
         loading: false,
         userData: {
-          repositories: false,
+          repos: false,
```
Comment (GrayStrider) [inline app/components/ReposList/tests/index.test.tsx:35]: we should either use `repos` everywhere, or `repositories` everywhere. I propose the shorter variant

File: app/containers/App/reducer.ts [hunk 4]
```diff
@@ -7,7 +7,7 @@ export const initialState: ContainerState = {
   error: false,
   currentUser: '',
   userData: {
-    repositories: [],
+    repos: [],
```
Comment (GrayStrider) [inline app/containers/App/reducer.ts:10]: https://github.com/Can-Sahin/react-boilerplate-typescript/pull/45/files#r339378580

File: app/containers/App/reducer.ts [hunk 5]
```diff
@@ -23,7 +23,7 @@ function appReducer(
         loading: true,
         error: false,
         userData: {
-          repositories: [],
+          repos: [],
```
Comment (GrayStrider) [inline app/containers/App/reducer.ts:26]: https://github.com/Can-Sahin/react-boilerplate-typescript/pull/45/files#r339378580

File: app/utils/loadable.tsx [hunk 6]
```diff
@@ -9,7 +9,7 @@ const loadable = <T extends React.ComponentType<any>>(
 ) => {
   const LazyComponent = lazy(importFunc);
 
-  return (props: any) => (
+  return (props: React.ComponentProps<T>): JSX.Element => (
```
Comment (GrayStrider) [inline app/utils/loadable.tsx:12]: same, might not the be the most correct variant but it works

**Comment by GrayStrider** [comment]
Reviewed everything, the main question is do we use `repos` or `repositories`? Let's pick one @Can-Sahin

**Pull request merged**